### PR TITLE
Feature/events

### PR DESCRIPTION
**Pull request by nottherealalanturing**
## Describe your changes
Link the calculate function to the UI.  Users can now 

- Perform addition, subtraction, division, multiple and modulo operations
- UI gets updated automatically when user interacts with calculator

### What kind of change does this PR introduce?
Add events

### In which branch do you want the changes to merge?

dev.

### Documentation updated ?

Yes

### Checklist

- [ ] No ES Lint Error
- [ ] No Style Lint Error
- [ ] Unit test case update

###Screenshot
Screen Shot 2022-05-26 at 14 45 09

### CODE REVIEW

File: src/component/Calculator.js [hunk 1]
```diff
@@ -28,18 +29,42 @@ const operatorsArray = ['÷', '*', '-', '+', '='];
 class Calculator extends React.Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      total: 0,
+      next: null,
+      operation: null,
+    };
```
Comment (Cathella) [inline src/component/Calculator.js:36]: ✅ Use of the state and props 👍🏻

File: src/component/Calculator.js [hunk 2]
```diff
@@ -28,18 +29,42 @@ const operatorsArray = ['÷', '*', '-', '+', '='];
...
             const operator = operatorsArray.includes(value) ? 'operator' : '';
             return (
-              <button className={`calculatorBtn ${operator}`} type="button" key={value}>
+              <button
+                className={`calculatorBtn ${operator}`}
+                type="button"
+                key={value}
+                name={value}
+                onClick={this.handleClick}
+              >
```
Comment (Cathella) [inline src/component/Calculator.js:67]: ✅ Implemented the click event for all buttons 👍🏻

File: src/component/Calculator.js [hunk 3]
```diff
@@ -1,5 +1,6 @@
 import React from 'react';
 import './styles/Calculator.css';
+import calculate from '../logic/calculate';
```
Comment (Cathella) [inline src/component/Calculator.js:3]: ✅ The Calculator component should import the `calculate.js` file from the logic directory 👍🏻

**Review by Cathella — CHANGES_REQUESTED** [review]
## 🟡 Status: Changes Required 🛑

### Hi @nottherealalanturing 👋🏻, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## To Highlight 👏🏻

✅ No linter errors.
✅ You used correct GitHub Flow.

## Required Changes ♻️
_Check the comments under the review._

- [ ] The calculator application seems to work well but the multiplication operation does work well, when I try multiplying numbers when I press the equals button, it returns a blank page. Kindly fix this behavior please.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by hiromataba — APPROVED** [review]
# Status: Approved :tada: :tada: :tada: 
Hi @nottherealalanturing    :wave: :wave: 

200w  

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**